### PR TITLE
Integrate campaigns list

### LINKED ISSUE #30: Campaigns List Integration

### PR DESCRIPTION
**Pull request by zognin**
Sometimes the frontend shows error, that's because the backend takes too long to load and it times out.

Closes #30

### CODE REVIEW

File: frontend/pages/campaigns/index.tsx [hunk 1]
```diff
@@ -1,48 +1,35 @@
...
-};
-
-const sampleCampaigns: CampaignListData[] = Array(11).fill(sampleCampaign);
+import useSWR from 'swr';
+import { Nullable } from '../../types/utils';
+import api from '../../frontendApis';
+import CampaignsAPI from '../../frontendApis/campaigns';
+import CampaignListLoading from '../../components/campaigns/list/CampaignListLoading';
 
 const Campaigns = () => {
```
Comment (sivayogasubramanian) [inline frontend/pages/campaigns/index.tsx:16]: image

We can remove the box-shadow? If not looks quite weird.

File: frontend/pages/campaigns/index.tsx [hunk 2]
```diff
@@ -51,9 +38,24 @@ const Campaigns = () => {
       </Head>
 
       <Container sx={containerSx} component="main">
-        <CampaignSearch searchDrawerIsOpen={searchDrawerIsOpen} setSearchDrawerIsOpen={setSearchDrawerIsOpen} />
+        <CampaignSearch
```
Comment (sivayogasubramanian) [inline frontend/pages/campaigns/index.tsx:41]: image

image

Can we move the buttons down? Seems like is the convention for this type of forms.

**Review by sivayogasubramanian — APPROVED** [review]
LGTM! Functionality is good. Some UI changes requested.

File: frontend/pages/campaigns/index.tsx [hunk 3]
```diff
@@ -51,9 +38,24 @@ const Campaigns = () => {
...
+
+        {error && (
+          <Stack spacing={2}>
+            <Typography variant="h1">Error</Typography>
+            <Typography variant="h2">That is all we know right now.</Typography>
+          </Stack>
+        )}
 
-        <CampaignList campaigns={sampleCampaigns} />
+        {campaigns && !error && <CampaignList campaigns={campaigns} />}
```
[the remark this answers is not in the document]
Comment (zognin) [reply frontend/pages/campaigns/index.tsx:58]: HY's PR is doing it. Oh yea also the contribute button should go to the other version of the redeem page (donate directly without redeeming a coupon), which we haven't done so im thinking of removing it before we do user testing tmr.

**Pull request merged**